src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
```

Comment:
PullProofAI

There is a potential security vulnerability in the `LoggerMiddleware` where the JWT token is being decoded without proper validation, which can lead to security risks if the token is tampered with or invalid.

```diff
  use(req: Request, res: Response, next: NextFunction) {
    const token = req.headers?.authorization?.split(' ')[1] || '';
    const userDetail = this.jwt.decode(token) || {};
```

#### Suggested Improvement

It's crucial to validate the JWT token before decoding it to ensure its integrity and authenticity. Consider implementing proper error handling and validation checks before decoding the token to prevent security vulnerabilities. Additionally, you may want to verify the token signature and expiration to enhance security.

By adding validation steps, you can mitigate potential security risks associated with decoding potentially malicious or tampered tokens.

Ensure to update the `LoggerMiddleware` implementation with robust token validation logic to enhance security practices within the application.


<!-- This is an auto-generated comment by PullProofAI -->